spec/immutable_struct_spec.rb

```diff
@@ -111,6 +111,26 @@ def self.from_array(array)
     end
   end
 
+  describe "coercion" do
+    let(:klass) { ImmutableStruct.new(:lolwat) }
+
+    it "is a noop when value is already the defined type" do
+      value = klass.new
+      new_value = klass.coerce(value)
+      expect(new_value).to be(value)
```

Comment:
Does `be` check object equality, e.g. that `object_id` matches? If so 👍 , if not, can/should we check that?

Reply:
I assumed that it does, but I'll verify that matcher's behavior.

Reply:
According to [the docs](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers) the `be` matcher is for "object identity" using `equal?`. I think that's right? https://stackoverflow.com/questions/7156955/whats-the-difference-between-equal-eql-and